### ISSUE TITLE
trained onnx model usage with pytorch

### ISSUE THREAD

**Issue by rvxfahim** [ENOT-AutoDL/onnx2torch#164]
I have trained a resnet18 model in Matlab using the dnn machine learning toolbox and saved the model in onnx format. Now I am trying to use it but the predictions are wrong, in matlab the model is predicting correctly using its own variable which is mat. I think I can't figure out how to load this model properly.

```
model_path = "/kaggle/input/onnxmodel13/resnet18_classification.onnx"
original_model = onnx.load(model_path)

converted_model = version_converter.convert_version(original_model,13)

pytorch_model = convert(converted_model)
summary(pytorch_model.cuda(), (3, 224, 224))

image_path = '/kaggle/input/classb/Priority/4df9d3ae-ed9a-11ed-b020-a46bb6069316.jpg'

image = Image.open(image_path)


transform = transforms.Compose([
    transforms.Resize((224, 224)),  # Resize to match the input size expected by the model
    transforms.ToTensor(),  # Convert PIL Image to PyTorch tensor
    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])  # Normalize pixel values
])
input_image = transform(image)


device = torch.device('cuda')
input_tensor = input_image.unsqueeze(0).to(device)


pytorch_model.eval()
with torch.no_grad():
    output = pytorch_model(input_tensor)

print(output)

_, predicted_idx = torch.max(output, 1)
predicted_class = predicted_idx.item()

class_names = ['Charlock', 'Priority', 'Stop', 'Sugarbeet']

predicted_class_name = class_names[predicted_class]
print("Predicted Class:", predicted_class_name)
print("Prediction:", output[0, predicted_class].item())
```

**Comment by agoncharenko1992** [comment]
Hello there!

Could you check whether you pre-process your input data correctly? There are a lot of hidden problems: channel order (RGB, BGR), input tensor normalization and so on.
